monai/losses/sure_loss.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Callable, Optional
+
+import torch
+import torch.nn as nn
+from torch.nn.modules.loss import _Loss
+
+
+def complex_diff_abs_loss(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+    """
+    First compute the difference in the complex domain,
+    then get the absolute value and take the mse
+
+    Args:
+        x, y - B, 2, H, W real valued tensors representing complex numbers
+                or  B,1,H,W complex valued tensors
+    Returns:
+        l2_loss - scalar
+    """
+    if not x.is_complex():
+        x_ = torch.view_as_complex(x.permute(0, 2, 3, 1).contiguous())
+    else:
+        x_ = x
+    if not y.is_complex():
+        y_ = torch.view_as_complex(y.permute(0, 2, 3, 1).contiguous())
+    else:
+        y_ = y
+    diff = x_ - y_
+    l2_loss = nn.functional.mse_loss(torch.abs(diff), torch.zeros_like(torch.abs(diff)), reduction="mean")
+    return l2_loss
```

Comment:
```suggestion
    if not x.is_complex():
        x = torch.view_as_complex(x.permute(0, 2, 3, 1).contiguous())
    if not y.is_complex():
        y = torch.view_as_complex(y.permute(0, 2, 3, 1).contiguous())
        
    diff = torch.abs(x - y)
    return nn.functional.mse_loss(diff, torch.zeros_like(diff), reduction="mean")
```

I think it's fine to simplify a bit like this.